api/worker/handlers.go

```diff
@@ -0,0 +1,99 @@
+package worker
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/ufcg-lsd/arrebol-pb/api"
+	"github.com/ufcg-lsd/arrebol-pb/crypto"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const SIGNATURE_HEADER string = "SIGNATURE";
+
+type WorkerSpec struct {
+	ID      string  `json:"ID"`
+	VCPU    float32 `json:"vCPU"`
+	RAM     uint32  `json:"RAM"` //Megabytes
+	QueueId string  `json:"QueueId, omitempty"`
+}
+
+type Token struct {
+	WorkerId string `json:"WorkerId"`
+	QueueId string  `json:"QueueId"`
+	VCPU	float32 `json:"vCPU"`
+	RAM     uint32  `json:"RAM"`
+}
+
+func (t Token) getRaw() string {
+	const separator string = "#"
+	builder := strings.Builder{}
+	builder.WriteString(t.WorkerId)
+	builder.WriteString(separator)
+	builder.WriteString(t.QueueId)
+	builder.WriteString(separator)
+	builder.WriteString(fmt.Sprintf("%.2f", t.VCPU))
+	builder.WriteString(separator)
+	builder.WriteString(fmt.Sprint(t.RAM))
+	return builder.String()
+}
+
+func (a *WorkerApi) AddWorker(w http.ResponseWriter, r *http.Request) {
+	err := a.verifySignature(r)
+	if err != nil {
+		api.Write(w, http.StatusUnauthorized, api.ErrorResponse{
+			Message: err.Error(),
+			Status:  http.StatusUnauthorized,
+		})
+		return
+	}
+
+	var workerSpec WorkerSpec
+	_ = json.NewDecoder(r.Body).Decode(&workerSpec)
```

Comment:
check this error and every you haven't checked...

Reply:
Done.